src/ThirdParty/ThirdParty.service.ts

```diff
@@ -0,0 +1,15 @@
+import { thirdPartyAPI } from '../ethereum/api/thirdParty'
+import { ItemCuration } from '../Curation/ItemCuration'
+import { ThirdParty } from './ThirdParty.types'
+
+export class ThirdPartyService {
+  static async getThirdPartyAvailableSlots(
+    thirdPartyId: ThirdParty['id']
+  ): Promise<number> {
+    const maxItems = await thirdPartyAPI.fetchMaxItemsByThirdParty(thirdPartyId)
+    const itemCurationsCount = await ItemCuration.getItemCurationCountByThirdPartyId(
+      thirdPartyId
+    )
```

Comment:
You could, in theory, `Promise.all` these methods. Not sure how much sense it makes tho!

Reply:
YES! They can be calculated in parallel, so using `Promise.all` here is a really good idea. Updated!